src/main/java/uk/gov/hmcts/reform/migration/repository/ElasticSearchRepository.java

```diff
@@ -0,0 +1,85 @@
+package uk.gov.hmcts.reform.migration.repository;
+
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Repository;
+import uk.gov.hmcts.reform.authorisation.generators.AuthTokenGenerator;
+import uk.gov.hmcts.reform.ccd.client.CoreCaseDataApi;
+import uk.gov.hmcts.reform.ccd.client.model.CaseDetails;
+import uk.gov.hmcts.reform.ccd.client.model.SearchResult;
+import uk.gov.hmcts.reform.migration.query.ElasticSearchQuery;
+
+import java.util.ArrayList;
+import java.util.List;
+
+@Repository
+@Slf4j
+public class ElasticSearchRepository {
+
+    private final CoreCaseDataApi coreCaseDataApi;
+
+    private final AuthTokenGenerator authTokenGenerator;
+
+    private final int querySize;
+
+    @Autowired
+    public ElasticSearchRepository(CoreCaseDataApi coreCaseDataApi,
+                                   AuthTokenGenerator authTokenGenerator,
+                                   @Value("${case-migration.elasticsearch.querySize}") int querySize) {
+        this.coreCaseDataApi = coreCaseDataApi;
+        this.authTokenGenerator = authTokenGenerator;
+        this.querySize = querySize;
+    }
+
+    public List<CaseDetails> findCaseByCaseType(String userToken, String caseType) {
+        ElasticSearchQuery elasticSearchQuery = ElasticSearchQuery.builder()
+            .initialSearch(true)
+            .size(querySize)
+            .build();
+
+        log.info("Processing the Case Migration search for case type {}.", caseType);
+        String authToken = authTokenGenerator.generate();
+
+        SearchResult searchResult = coreCaseDataApi.searchCases(userToken,
+                                                                authToken,
+                                                                caseType, elasticSearchQuery.getQuery()
+        );
+
+        List<CaseDetails> caseDetails = new ArrayList<>();
+
+        if (searchResult.getTotal() > 0) {
```

Comment:
@kiran-yenigala-hmcts We can add null check for searchResult and subsequentSearchResult  as it is getting the values from external (data-store) API and we never know when things change.